catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-converter/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_1_0/catalog/converter/impl/AbstractFeatureConverterWfs11.java

```diff
@@ -0,0 +1,215 @@
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactoryConfigurationError;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import org.codice.ddf.libs.geo.util.GeospatialUtil;
+import org.codice.ddf.platform.util.XMLUtils;
+import org.codice.ddf.spatial.ogc.catalog.common.converter.XmlNode;
+import org.codice.ddf.spatial.ogc.wfs.catalog.converter.FeatureConverter;
+import org.codice.ddf.spatial.ogc.wfs.catalog.converter.impl.AbstractFeatureConverter;
+import org.codice.ddf.spatial.ogc.wfs.catalog.mapper.MetacardMapper;
+import org.codice.ddf.spatial.ogc.wfs.v1_1_0.catalog.common.Wfs11Constants;
+import org.geotools.xml.Configuration;
+import org.geotools.xml.Parser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+abstract class AbstractFeatureConverterWfs11 extends AbstractFeatureConverter
+    implements FeatureConverter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractFeatureConverterWfs11.class);
+
+  private static final String XML_PARSE_FAILURE = "Failed to parse GML based XML into a Document.";
+
+  private static final String CREATE_TRANSFORMER_FAILURE = "Failed to create Transformer.";
+
+  private static final String GML_FAILURE = "Failed to transform GML.\n";
+
+  private static final XMLUtils XML_UTILS = XMLUtils.getInstance();
+
+  AbstractFeatureConverterWfs11(String srs) {
+    this.setSrs(srs);
+  }
+
+  AbstractFeatureConverterWfs11(MetacardMapper metacardMapper) {
+    super(metacardMapper);
+  }
+
+  protected Serializable getValueForMetacardAttribute(
+      AttributeFormat attributeFormat, HierarchicalStreamReader reader) {
+
+    Serializable ser = null;
+    switch (attributeFormat) {
+      case BOOLEAN:
+        ser = Boolean.valueOf(reader.getValue());
+        break;
+      case DOUBLE:
+        ser = Double.valueOf(reader.getValue());
+        break;
+      case FLOAT:
+        ser = Float.valueOf(reader.getValue());
+        break;
+      case INTEGER:
+        ser = Integer.valueOf(reader.getValue());
+        break;
+      case LONG:
+        ser = Long.valueOf(reader.getValue());
+        break;
+      case SHORT:
+        ser = Short.valueOf(reader.getValue());
+        break;
+      case XML:
+      case STRING:
+        ser = reader.getValue();
+        break;
+      case GEOMETRY:
+        XmlNode node = new XmlNode(reader);
+        String geometryXml = node.toString();
+        Geometry geo;
+
+        geo = readGml(geometryXml);
+
+        LOGGER.debug("coordinateOrder = {}", coordinateOrder);
+        if (GeospatialUtil.LAT_LON_ORDER.equals(coordinateOrder)) {
+          swapCoordinates(geo);
+        }
+
+        if (geo != null) {
+          WKTWriter wktWriter = new WKTWriter();
```

Comment:
it may be worthwhile to make this a global, instead of creating a new WKTWriter for every invocation of `getValueForMetacardAttribute`

Reply:
I'll make it a thread-local